pkg/http/api.go

```diff
@@ -1454,27 +1523,42 @@ func (a *api) onDirectMessage(reqCtx *fasthttp.RequestCtx) {
 	cancel()
 }
 
-// findTargetID tries to find ID of the target service from the following three places:
+// findTargetID tries to find ID of the target service from the following four places:
 // 1. {id} in the URL's path.
 // 2. Basic authentication, http://dapr-app-id:<service-id>@localhost:3500/path.
 // 3. HTTP header: 'dapr-app-id'.
+// 4. HTTP Endpoint baseURL override, http://localhost:3500/v1.0/invoke/<overwritten baseURL so targetID here>/method/<method>
 func (a *api) findTargetID(reqCtx *fasthttp.RequestCtx) string {
-	if id := reqCtx.UserValue(idParam); id == nil {
-		if appID := reqCtx.Request.Header.Peek(daprAppID); appID == nil {
-			if auth := reqCtx.Request.Header.Peek(fasthttp.HeaderAuthorization); auth != nil &&
-				strings.HasPrefix(string(auth), "Basic ") {
-				if s, err := base64.StdEncoding.DecodeString(strings.TrimPrefix(string(auth), "Basic ")); err == nil {
-					pair := strings.Split(string(s), ":")
-					if len(pair) == 2 && pair[0] == daprAppID {
-						return pair[1]
-					}
-				}
+	if id := reqCtx.UserValue(idParam); id != nil {
+		return id.(string)
+	}
+
+	if appID := reqCtx.Request.Header.Peek(daprAppID); appID != nil {
+		return string(appID)
+	}
+
+	if auth := reqCtx.Request.Header.Peek(fasthttp.HeaderAuthorization); auth != nil &&
+		strings.HasPrefix(string(auth), "Basic ") {
+		if s, err := base64.StdEncoding.DecodeString(strings.TrimPrefix(string(auth), "Basic ")); err == nil {
+			pair := strings.Split(string(s), ":")
+			if len(pair) == 2 && pair[0] == daprAppID {
+				return pair[1]
 			}
-		} else {
-			return string(appID)
 		}
-	} else {
-		return id.(string)
+	}
+
+	uri := string(reqCtx.URI().Path())
+	if strings.HasPrefix(uri, "/v1.0/invoke/") {
+		parts := strings.Split(uri, "/")
+		// Example: http://localhost:3500/v1.0/invoke/http://api.github.com/method/<method>
```

Comment:
I want to reconfirm that URI only has three parts, and do we need to check it in the code first?

Reply:
We updated my branch to include the configurability of the base address, so this should be resolved now. Thanks for catching!